src/components/DatePicker/OverlayComponent/OverlayComponent.tsx

```diff
@@ -0,0 +1,55 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React from 'react';
+import useTheme from '../../../hooks/useTheme';
+import { optionStyle } from '../DatePicker.style';
+import { ExtraOption } from '../DatePicker';
+import { DayPickerInputProps } from 'react-day-picker';
+import omit from 'lodash/omit';
+
+type Props = {
+  selectedOption: string;
+  setSelectedOption: Function;
+  isRangePicker: boolean;
+  extraOptions: ExtraOption[];
+};
+const OverlayComponent: React.FC<Props & DayPickerInputProps> = ({
+  classNames,
+  selectedOption,
+  setSelectedOption,
+  isRangePicker,
+  extraOptions = [],
+  children,
+  ...props
+}) => {
+  const theme = useTheme();
+
+  return (
+    <div
+      className={classNames?.overlayWrapper}
+      css={{ marginTop: 3 }}
+      {...omit(props, ['onKeyUp', 'onClick'])}
```

Comment:
Why are you removing this props from the div element?
You can assign this to a method to improve the readability or add an explanatory comment.